HotelSearchingListingBookingEngine.Core/Parsers/HotelSearchRQParser.cs

```diff
@@ -156,7 +156,7 @@ private DateTimeSpan getStayPeriod(DateTime checkInDate, DateTime checkOutDate)
 
         private int getMinimumRoomsRequired(int adultsCount, int childrensCount)
         {
-            return (adultsCount / 2 + childrensCount / 2);
+            return (adultsCount / 2 + childrensCount / 2 + 1);
```

Comment:
logic for calculating rooms based on number of guests can be improved
Please try using a better way